src/boss-utility.ts

```diff
@@ -1,42 +1,45 @@
 import { loadJson } from './load-json.js';
+import { camelize } from "./util.js";
+import { FORMATTED_BOSS_NAMES } from "osrs-json-hiscores";
 const constants = loadJson('static/constants.json');
 
 /**
- * Boss name with capitalization
+ * Boss name with capitalization and spacing
  * @typedef {string} BossName
  */
 
 /**
-  * Boss name in all lowercase
+  * Boss name in camelcase with only alphanumeric characters
   * @typedef {string} BossID
   */
 
-const validBossNames: Set<string> = new Set<string>(constants.bosses);
-const validBossIDs: Set<string> = new Set<string>(constants.bosses.map(b => sanitizeBossName(b)));
+const validBossNames = new Set(Object.values(FORMATTED_BOSS_NAMES));
+const validBossIDs = new Set(Object.keys(FORMATTED_BOSS_NAMES));
 
 export function toSortedBosses(bosses: string[]): string[] {
     const bossSubset = new Set(bosses);
     return [...validBossIDs].filter(bossID => bossSubset.has(bossID));
 };
 
 /**
- * Returns lowercase of boss name
+ * Converts boss name to camelcase and removes non-alphanumeric
+ * characters
  * @param {BossName} bossName
  * @returns {BossID}
  */
  export function sanitizeBossName(bossName: string): string {
-    const bossID = bossName.toLowerCase();
+    const bossID = camelize(bossName).replace(/\W/g, '');
     return bossID;
 }
 
 /**
- * Gets boss name from ID
+ * Gets boss name from ID or returns string "Unknown"
  * @param {BossID} bossID
  * @returns {BossName}
  */
  export function getBossName(bossID: string): string {
-    const bossName = constants.bosses.find(b => sanitizeBossName(b) === bossID);
-    return bossName;
+    const bossName = Object.values(FORMATTED_BOSS_NAMES).find((b: string) => sanitizeBossName(b) === bossID);
```

Comment:
This function seems really complicated. I don't understand why you can't just do:
```
return FORMATTED_BOSS_NAMES[bossID] ?? 'Unknown'
```

Reply:
yeah 1 sec

Reply:
https://github.com/evanw555/scape-bot/pull/19/commits/6fa9e9de3aeec3cbb90f1846ce0041ba4c8406a5